fuzz/src/invoice_request_deser.rs

```diff
@@ -134,6 +134,12 @@ fn build_response<T: secp256k1::Signing + secp256k1::Verification>(
 	)
 	.unwrap();
 
+	if let Ok(verified) = invoice_request.clone().verify_using_metadata(&expanded_key, secp_ctx) {
+		// Previously we had a panic where we'd truncate the payer note possibly cutting a Unicode
+		// character in two here, so try to fetch fields if we can validate.
+		let _ = verified.fields();
```

Comment:
Huh, so looking earlier in `build_response`, we are already manually building the `InvoiceRequestFields` only not truncating. We should probably move this there instead.